### PR TITLE
Improve `packVNNIMatmulOp`

### PR DESCRIPTION
**Pull request by chelini**
Checking the rank is not the right way. A better approach is to check the affine map and bail out if the map is already in VNNI layout.

### CODE REVIEW

File: lib/TPP/Transforms/ToBlockLayoutAndBack.cpp [hunk 1]
```diff
@@ -401,6 +401,41 @@ mlir::linalgx::packMatmulOp(RewriterBase &rewriter,
...
+  AffineExpr vnniDim = results.back();
+  auto dimExpr = dyn_cast<AffineDimExpr>(vnniDim);
+  if (!dimExpr ||
+      iteratorTypes[dimExpr.getPosition()] != utils::IteratorType::reduction) {
+    return false;
+  }
+
+  for (auto result : results) {
+    if (auto blockeDim = dyn_cast<AffineBinaryOpExpr>(result)) {
+      if (blockeDim.getKind() == AffineExprKind::FloorDiv) {
```
Comment (rengolin) [inline lib/TPP/Transforms/ToBlockLayoutAndBack.cpp:422]:
```suggestion
    auto blockeDim = dyn_cast<AffineBinaryOpExpr>(result);
    if (!blockedDim)
      continue;
    if (blockeDim.getKind() != AffineExprKind::FloorDiv)
      continue;
```

File: test/BF16/brgemm-vnni.mlir [hunk 2]
```diff
@@ -1,14 +1,120 @@
...
+// CHECK-LABEL: brgemm
+// CHECK-SAME:  %[[ARG0:.+]]: tensor<32x4x4xbf16>, %[[ARG1:.+]]: tensor<32x4x4xbf16>, 
+// CHECK-SAME:  %[[ARG2:.+]]: tensor<4x4xbf16>
 // CHECK: %[[EMPTY:.+]] = tensor.empty() : tensor<32x2x4x2xbf16>
 // CHECK: %[[PACK:.+]] = tensor.pack %[[ARG1]]
-// CHECK-SAME:  inner_dims_pos = [1] inner_tiles = [2] into %[[EMPTY]] : tensor<32x4x4xbf16> -> tensor<32x2x4x2xbf16>
-// CHECK: tpp.brgemm (%[[ARG0]] : tensor<32x4x4xbf16>, %[[PACK]] : tensor<32x2x4x2xbf16>, %[[ARG2]] : tensor<4x4xbf16>) -> (tensor<4x4xbf16>)
+// CHECK-SAME:  inner_dims_pos = [1] inner_tiles = [2] into %[[EMPTY]] 
+// CHECK-SAME:  : tensor<32x4x4xbf16> -> tensor<32x2x4x2xbf16>
+// CHECK: tpp.brgemm (%[[ARG0]] : tensor<32x4x4xbf16>, %[[PACK]] : tensor<32x2x4x2xbf16>,
```
Comment (rengolin) [inline test/BF16/brgemm-vnni.mlir:17]: We really should not be adding more tpp dialect tests...

Comment (chelini) [reply test/BF16/brgemm-vnni.mlir:17]: if you wait for another patch all of these will disappear.

Comment (chelini) [reply test/BF16/brgemm-vnni.mlir:17]: "all of these" = tpp.brgemm

**Pull request merged**